app/scripts/modules/core/src/cloudProvider/cloudProvider.registry.ts

```diff
@@ -25,30 +26,30 @@ class Providers {
     // The tests depend on this behavior, but maybe something else does as well.
     this.providers = without(
       this.providers,
-      this.providers.find(p => p.cloudProvider === cloudProvider && p.config.providerVersion === config.providerVersion)
+      this.providers.find(p => p.cloudProvider === cloudProvider && p.config.skin === config.skin)
     ).concat([{ cloudProvider, config }]);
   }
 
-  public get(cloudProvider: string, providerVersion?: string): ICloudProviderConfig {
-    if (providerVersion) {
-      const provider = this.providers.find(p => p.cloudProvider === cloudProvider && p.config.providerVersion === providerVersion);
-      return provider ? provider.config : this.getDefaultConfig(cloudProvider);
+  public get(cloudProvider: string, skin?: string): ICloudProviderConfig {
+    if (skin) {
```

Comment:
Why are we looking things up by their `skin`? Isn't that a property of what you want to find? Or is it overloading `providerVersion`?